### PR TITLE
[CLOUDTRUST-2264] Add GetUser (by id) in kyc API

### PR DESCRIPTION
**Pull request by fperot74**
Coverage should decrease but I don't want to write unit tests for deprecated methods that will be removed soon.

### CODE REVIEW

File: api/kyc/swagger-api_kyc.yaml [hunk 1]
```diff
@@ -50,9 +50,29 @@ paths:
...
+          type: string
+      responses:
+        200:
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        403:
+          description: No permission to call this operation
```
Comment (bsoniam) [inline api/kyc/swagger-api_kyc.yaml]: There is no more 404 possible?

Comment (fperot74) [reply api/kyc/swagger-api_kyc.yaml]: I'ld like so but Keycloak doesn't return 404 when users are not found

File: cmd/keycloakb/keycloak_bridge.go [hunk 2]
```diff
@@ -631,9 +631,10 @@ func main() {
 		kycComponent = kyc.MakeAuthorizationRegisterComponentMW(registerRealm, log.With(kycLogger, "mw", "endpoint"), authorizationManager)(kycComponent)
 
 		kycEndpoints = kyc.Endpoints{
-			GetActions:   prepareEndpoint(kyc.MakeGetActionsEndpoint(kycComponent), "register_get_actions", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
-			GetUser:      prepareEndpoint(kyc.MakeGetUserEndpoint(kycComponent), "get_user", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
-			ValidateUser: prepareEndpoint(kyc.MakeValidateUserEndpoint(kycComponent), "validate_user", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
+			GetActions:    prepareEndpoint(kyc.MakeGetActionsEndpoint(kycComponent), "register_get_actions", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
+			GetUser:       prepareEndpoint(kyc.MakeGetUserEndpoint(kycComponent), "get_user", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
+			GetUserByName: prepareEndpoint(kyc.MakeGetUserByNameEndpoint(kycComponent), "get_user_by_name", influxMetrics, kycLogger, tracer, rateLimit["kyc"]),
```
Comment (harture) [inline cmd/keycloakb/keycloak_bridge.go]: GetUserByUsername ?

File: pkg/kyc/authorization.go [hunk 3]
```diff
@@ -63,7 +63,7 @@ func (c *authorizationComponentMW) GetActions(ctx context.Context) ([]apikyc.Act
 	return c.next.GetActions(ctx)
 }
 
-func (c *authorizationComponentMW) GetUser(ctx context.Context, username string) (apikyc.UserRepresentation, error) {
+func (c *authorizationComponentMW) GetUserByName(ctx context.Context, username string) (apikyc.UserRepresentation, error) {
 	var action = KYCGetUser.String()
 	var targetRealm = c.realmName
 	var groupID = RegistrationOfficer
```
Comment (harture) [inline pkg/kyc/authorization.go]: We should call CheckAuthorizationOnTargetUser. 
groupID is the one of the target so it seems incorrect there.

File: pkg/kyc/authorization.go [hunk 4]
```diff
@@ -72,7 +72,19 @@ func (c *authorizationComponentMW) GetUser(ctx context.Context, username string)
...
-	return c.next.GetUser(ctx, username)
+	return c.next.GetUserByName(ctx, username)
+}
+
+func (c *authorizationComponentMW) GetUser(ctx context.Context, userID string) (apikyc.UserRepresentation, error) {
+	var action = KYCGetUser.String()
+	var targetRealm = c.realmName
+	var groupID = RegistrationOfficer
+
+	if err := c.authManager.CheckAuthorizationOnTargetGroup(ctx, action, targetRealm, groupID); err != nil {
```
Comment (harture) [inline pkg/kyc/authorization.go]: Same ass before

File: cmd/keycloakb/keycloak_bridge.go (hunk 2 above)
Comment (fperot74) [reply cmd/keycloakb/keycloak_bridge.go]: Done

**Pull request merged**